src/main/java/seedu/task/logic/parser/AddParser.java

```diff
@@ -45,24 +52,22 @@ public Command prepare(String args){
         final Matcher taskMatcher = TASK_DATA_ARGS_FORMAT.matcher(args.trim());
         final Matcher eventMatcher = EVENT_DATA_ARGS_FORMAT.matcher(args.trim());
         
+        ArgumentTokenizer argsTokenizer = new ArgumentTokenizer(descriptionPrefix, deadlinePrefix, durationPrefix);
```

Comment:
I thought of that too, but there isn't really a flag present to capture if something is a task or an event. The main criteria to distinguish is if /from is present. We can refine the regex to detect if /from is present. Otherwise, I can't think of a way to improve on it. You have any idea on how to do this @TiankaiChen .
The argumentstokenizer was given by Prof, @githubxxcc , I didn't code it out. Just adopted it hha